deepecohab/antenna_analysis/activity.py

```diff
@@ -176,36 +61,37 @@ def calculate_time_spent_per_position(
     """
     cfg = auxfun.read_config(cfp)
     
-    results_path = Path(cfg['project_location']) / 'results' / 'results.h5'
+    results_path = Path(cfg['project_location']) / 'results'
     key='time_per_position'
     
-    time_per_position = None if overwrite else auxfun.load_ecohab_data(cfp, key, verbose=False)
+    time_per_position_lf = None if overwrite else auxfun.load_ecohab_data(cfp, key, verbose=False)
     
-    if isinstance(time_per_position, pd.DataFrame):
-        return time_per_position
-    
-    tunnels = cfg['tunnels']
-    
-    df = auxfun.load_ecohab_data(cfg, key='main_df')
-    padded_df = create_padded_df(cfp, df, overwrite=overwrite)
-    
-    # Map directional tunnel position to non-directional
-    mapper = padded_df.position.isin(tunnels.keys())
-    padded_df.position = padded_df.position.astype(str)
-    padded_df.loc[mapper, 'position'] = padded_df.loc[mapper, 'position'].map(tunnels).values
-    
-    # Calculate time spent per position per phase
-    time_per_position = (
-        padded_df
-        .groupby(['phase', 'day', 'phase_count', 'position', 'animal_id'], observed=True)['timedelta']
-        .sum()
-        .unstack('animal_id')
-    )
+    if isinstance(time_per_position_lf, pl.LazyFrame):
+        return time_per_position_lf
     
+    animal_ids = list(cfg['animal_ids'])
+
+    padded_lf = auxfun.load_ecohab_data(cfg, key='padded_df')
+
+    # Map directional tunnel position to non-directional and calculate time spent per position per phase
+    padded_lf = auxfun.remove_tunnel_directionality(padded_lf, cfg)
+
+    group_cols = ["phase", "day", "phase_count", "position"]
+
+    time_per_position_lf = padded_lf.group_by(
+            group_cols
+        ).agg(
+            auxfun.get_agg_expression(
+                value_col="timedelta",
+                animal_ids=animal_ids,
+                agg_fn=lambda e: e.sum(),
+            )
+        ).sort(group_cols)
+
     if save_data:
-        time_per_position.to_hdf(results_path, key=key, mode='a', format='table')
+        time_per_position_lf.sink_parquet(results_path / f"{key}.parquet", compression='lz4')
     
-    return time_per_position
+    return time_per_position_lf
 
 def calculate_visits_per_position(
```

Comment:
This should be combined with the previous one, and since it's long format the group cols should have `animal_id` and the 
```python
df.group_by(
        ["phase", "day", "phase_count", "position", "animal_id"]
    ).agg([
        pl.col("timedelta").sum().alias("time_spent"),
        pl.col("timedelta").count().alias("visits")
    ])
```